### DISCUSSION TITLE
Is there any way to specify the indent of text in a cell?

### DISCUSSION THREAD

**Discussion by iambic69** [dfinke/ImportExcel#1573] [Q&A]
<img width="55" alt="image" src="https://github.com/dfinke/ImportExcel/assets/10015793/e821ab0f-09db-401d-9564-a2ada08b6eac">

ImportExcel has heaps of options to control formatting, but **indent** seems to be missing, or am I the one missing something?

**Comment by dfinke** [comment]
```powershell
$data = ConvertFrom-Csv @"
Region,State,Units,Price
West,Texas,927,923.71
North,Tennessee,466,770.67
East,Florida,520,458.68
East,Maine,828,661.24
West,Virginia,465,053.58
North,Missouri,436,235.67
South,Kansas,214,992.47
North,North Dakota,789,640.72
South,Delaware,712,508.55
"@

$xlfile = ".\test.xlsx"
Remove-Item $xlfile -ErrorAction SilentlyContinue

$xlpkg = $data | Export-Excel $xlfile -PassThru

Set-ExcelRange -Worksheet $xlpkg.Sheet1 -Range 'A:A' -HorizontalAlignment Center
Set-ExcelRange -Worksheet $xlpkg.Sheet1 -Range 'B:B' -HorizontalAlignment Right

Close-ExcelPackage $xlpkg -Show
```

## Horizontal Alignment Options

![image](https://github.com/dfinke/ImportExcel/assets/67258/3b2f4ab9-8e63-4cc7-b4e8-64e38121849b)

**Reply by iambic69** [reply]
Thanks for your swift reply, Doug.  That covers cell alignment, but what I'm looking for is cell **_indent_** (set using the Increase/Decrease Indent buttons in the Excel ribbon).  Excel seems to store that in a style.

```powershell

$xlfile = ".\test.xlsx"
$excel = Open-ExcelPackage $xlfile
$id = $excel.Sheet1.Cells["A3"].StyleID
$excel.Workbook.Styles.CellXfs[$id]
``` 

[test.xlsx](https://github.com/dfinke/ImportExcel/files/14531061/test.xlsx)

<img width="78" alt="image" src="https://github.com/dfinke/ImportExcel/assets/10015793/fc5b26c8-657a-41e3-998f-67c70d706806">


Outputs:

<img width="398" alt="image" src="https://github.com/dfinke/ImportExcel/assets/10015793/dc95cce5-c6e8-4796-9ba0-f3d33b3cfa92">

I can't see any options to control the indent for a cell when exporting.

**Comment by dfinke** [comment]
that's diff than `outline`?

Nothing specific in the PowerShell functions to do that.

try:

`$excel.Workbook.Styles.CellXfs[$id] | get-member` 

See if there is a `{get; set}` for `indent` indicating a r/w property. There may be something like `set_indent` that lets you set the value.

**Reply by iambic69** [reply]
Thanks.  Yes, different to outline, and yes, there is a setter.  I wasn't sure how the whole styles thing worked, but I see that the underlying OfficeOpenXml library must do a clever copy-or-match-existing-on-write thing.  Indent can be set just by doing:

`$excel.Sheet1.Cells["A5"].Style.Indent = 5`

... which creates a new style in `Workbook.Styles.CellXfs`, if needed.  That gives me a workaround.

Adding it to ImportExcel would involve adding an `Indent` parameter to everywhere that `HorizontalAlignment` is supported:

- `New-ExcelStyle`
- `Set-ExcelRow`
- `Set-ExcelColumn`
- `Set-ExcelRange`

... and implementing it in `Set-ExcelRange` like this:

```powershell
            if ($PSBoundParameters.ContainsKey('Indent')) {
                $Range.Style.Indent = $Indent
            }
``` 

Does that sound about right?

**Reply by dfinke** [reply]
Thanks for diving in @iambic69!
That sounds right. I would need to review. Not sure if you are asking so you can do a PR or just update your local copy.

For a PR. Pester tests would need to be updated and added.

**Reply by iambic69** [reply]
I was thinking I might do a PR.  I'll see if I can find the time. :)

Thanks again!
